joint_trajectory_controller/test/joint_trajectory_controller_test.cpp

```diff
@@ -414,117 +558,104 @@ TEST_F(JointTrajectoryControllerTest, invalidMessages)
   {
     ActionGoal empty_goal;
     action_client->sendGoal(empty_goal);
-    ASSERT_TRUE(waitForState(action_client, SimpleClientGoalState::REJECTED, long_timeout));
-    EXPECT_EQ(action_client->getResult()->error_code, control_msgs::FollowJointTrajectoryResult::INVALID_JOINTS);
+    ASSERT_TRUE(waitForActionResult(action_client));
+    EXPECT_TRUE(checkActionGoalState(action_client, SimpleClientGoalState::REJECTED));
+    EXPECT_TRUE(checkActionResultErrorCode(action_client, FollowJointTrajectoryResult::INVALID_JOINTS));
   }
 }
 
 // Uninterrupted trajectory execution //////////////////////////////////////////////////////////////////////////////////
 
 TEST_F(JointTrajectoryControllerTest, topicSingleTraj)
 {
-  ASSERT_TRUE(initState());
-
   // Send trajectory
   traj.header.stamp = ros::Time(0); // Start immediately
   traj_pub.publish(traj);
-  ros::Duration wait_duration = traj.points.back().time_from_start + ros::Duration(0.5);
-  wait_duration.sleep(); // Wait until done
+  ASSERT_TRUE(waitForTrajectoryExecution());
+
+  auto timeout{getTrajectoryDuration(traj) + ros::Duration(TIMEOUT_TRAJ_EXECUTION_S)};
```

Comment:
{}{}

Reply:
Yep, I prepared a commit changing this already